interface.go

```diff
@@ -0,0 +1,97 @@
+package immutable
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/immutable/imx-core-sdk-golang/api"
+	"github.com/immutable/imx-core-sdk-golang/contracts"
+)
+
+var (
+	Sandbox = Environment{
+		BaseAPIPath:                 "https://api.ropsten.x.immutable.com",
+		RegistrationContractAddress: "0x6C21EC8DE44AE44D0992ec3e2d9f1aBb6207D864",
+		CoreContractAddress:         "0x4527BE8f31E2ebFbEF4fCADDb5a17447B27d2aef",
+		ChainID:                     big.NewInt(3), // Ropsten
+	}
+	Mainnet = Environment{
+		BaseAPIPath:                 "https://api.x.immutable.com",
+		RegistrationContractAddress: "0x72a06bf2a1CE5e39cBA06c0CAb824960B587d64c",
+		CoreContractAddress:         "0x5FDCCA53617f4d2b9134B29090C87D01058e27e9",
+		ChainID:                     big.NewInt(1),
+	}
+)
+
+type Config struct {
+	EthereumRPC string
+	APIConfig   *api.Configuration
+}
+
+type Environment struct {
+	BaseAPIPath                 string
+	RegistrationContractAddress string
+	CoreContractAddress         string
+	ChainID                     *big.Int
+}
+
+type Client struct {
+	Environment          Environment
+	EthClient            *ethclient.Client
+	RegistrationContract *contracts.Registration
+	CoreContract         *contracts.Core
+	api.TradesApi
+	api.OrdersApi
+	api.TransfersApi
+	api.DepositsApi
+	api.WithdrawalsApi
+	api.MintsApi
+	api.AssetsApi
+	api.UsersApi
+	api.EncodingApi
+	api.MetadataApi
+	api.ProjectsApi
+	api.TokensApi
+	api.CollectionsApi
+	api.BalancesApi
+}
```

Comment:
This will make interface implemented by members accessible by `Client` object. If there is a name clash for what ever reason this can break. 

When they are added as member vars, I prefer the clarity it provides in terms of the functions that are available for a particular endpoint.

Reply:
I've gone with how we structured it in the TS SDK - there are definitely tradeoffs, but we should be consistent. On the flattening, I actually prefer flatter - otherwise people have to already understand the schema (e.g. balances vs. tokens vs. assets)

Reply:
I feel the flat structure might confuse more. It will be more specific. if a user is only focusing on the relevant functions to get a particular job done like deposit or minting or assets, instead of understanding all the functions in flatter structure. No ambiguity when they are grouped accordingly.

Reply:
Yup, will try to maintain consistency. For limiting the public interface functions we might not want to expose them at top level. We only want to expose the limited interface.